packages/plugin-ext/src/main/browser/view/tree-view-widget.tsx

```diff
@@ -338,7 +479,33 @@ export class TreeViewWidget extends TreeViewWelcomeWidget {
             };
         }
 
-        if (node.tooltip && MarkdownString.is(node.tooltip)) {
+        const elementRef = React.createRef<HTMLDivElement & Partial<TooltipAttributes>>();
+        if (!node.tooltip && node instanceof ResolvableTreeViewNode) {
+            let configuredTip = false;
+            attrs = {
+                ...attrs,
+                'data-for': this.tooltipService.tooltipId,
+                onMouseEnter: async () => {
+                    if (configuredTip) {
+                        return;
+                    }
+                    if (!node.resolved) {
+                        await node.resolve(new CancellationTokenSource().token);
+                    }
+                    if (elementRef.current) {
+                        // Set the resolved tooltip. After an HTML element was created data-* properties must be accessed via the dataset
+                        elementRef.current.dataset.tip = MarkdownString.is(node.tooltip) ? this.markdownIt.render(node.tooltip.value) : node.tooltip;
+                        this.tooltipService.update();
+                        configuredTip = true;
+                        // Manually fire another mouseenter event to get react-tooltip to update the tooltip content.
+                        // Without this, the resolved tooltip is only shown after re-entering the tree item with the mouse.
+                        elementRef.current.dispatchEvent(new MouseEvent('mouseenter'));
+                    } else {
+                        console.error(`Could not set resolved tooltip for tree node '${node.id}' because its React Ref was not set.`);
+                    }
+                }
+            };
```

Comment:
Suggestion: I believe there is a clean way to use the token here. Something in the likes of this:
```suggestion
            let configuredTip = false;
            let source: CancellationTokenSource | undefined;
            attrs = {
                ...attrs,
                'data-for': this.tooltipService.tooltipId,
                onMouseLeave: () => source?.cancel(),
                onMouseEnter: async () => {
                    if (configuredTip) {
                        return;
                    }
                    source = new CancellationTokenSource();
                    const token = source.token;
                    if (!node.resolved) {
                        await node.resolve(token);
                    }
                    if (token.isCancellationRequested) {
                        return;
                    }
                    if (elementRef.current) {
                        // Set the resolved tooltip. After an HTML element was created data-* properties must be accessed via the dataset
                        elementRef.current.dataset.tip = MarkdownString.is(node.tooltip) ? this.markdownIt.render(node.tooltip.value) : node.tooltip;
                        this.tooltipService.update();
                        configuredTip = true;
                        // Manually fire another mouseenter event to get react-tooltip to update the tooltip content.
                        // Without this, the resolved tooltip is only shown after re-entering the tree item with the mouse.
                        elementRef.current.dispatchEvent(new MouseEvent('mouseenter'));
                    } else {
                        console.error(`Could not set resolved tooltip for tree node '${node.id}' because its React Ref was not set.`);
                    }
                }
            };
```